src/main/java/io/confluent/kafkarest/KafkaRestApplication.java

```diff
@@ -73,59 +82,33 @@ protected void setupInjectedResources(
   ) {
     config.register(new ZkExceptionMapper(appConfig));
 
-    if (zkUtils == null) {
-      zkUtils = ZkUtils.apply(
-          appConfig.getString(KafkaRestConfig.ZOOKEEPER_CONNECT_CONFIG),
-          30000,
-          30000,
-          JaasUtils.isZkSecurityEnabled()
-      );
-    }
-    if (mdObserver == null) {
-      mdObserver = new MetadataObserver(appConfig, zkUtils);
-    }
-    if (producerPool == null) {
-      producerPool = new ProducerPool(appConfig, zkUtils);
-    }
-    if (consumerManager == null) {
-      consumerManager = new ConsumerManager(appConfig, mdObserver);
-    }
-    if (simpleConsumerFactory == null) {
-      simpleConsumerFactory = new SimpleConsumerFactory(appConfig);
-    }
-    if (simpleConsumerManager == null) {
-      simpleConsumerManager =
-          new SimpleConsumerManager(appConfig, mdObserver, simpleConsumerFactory);
-    }
-    if (kafkaConsumerManager == null) {
-      kafkaConsumerManager = new KafkaConsumerManager(appConfig);
-    }
-
-    this.zkUtils = zkUtils;
-    context =
-        new Context(
-            appConfig,
-            mdObserver,
-            producerPool,
-            consumerManager,
-            simpleConsumerManager,
-            kafkaConsumerManager
-        );
+    KafkaRestContextProvider.initializeDefaultContext(zkUtils, appConfig, mdObserver, producerPool,
```

Comment:
Though not enforced, I believe we have a style preference for formatting long argument lists like the `ZKUtils.apply` call that was removed here (i.e. different indentation to what you have).